applications/dashboard/controllers/Api/ApplicationsApiController.php

```diff
@@ -0,0 +1,221 @@
+        $registrationMethod = strtolower($configuration->get('Garden.Registration.Method'));
+        if ($registrationMethod !== 'approval') {
+            throw new ClientException('The site is not configured for the approval registration method.');
+        }
+        $this->caseScheme = $caseScheme;
+        $this->userModel = $userModel;
+    }
+
+    /**
+     * Get a schema instance comprised of all available application fields.
+     *
+     * @return Schema Returns a schema object.
+     */
+    public function fullSchema() {
+        static $fullSchema;
+
+        if ($fullSchema === null) {
+            $fullSchema = Schema::parse([
+                'userID:i' => 'Unique user ID associated with the application.',
+                'email:s' => 'Email address associated with the application.',
+                'name:s' => 'Username on the application.',
+                'discoveryText:s' => 'Reason why you want to join.',
+                'status:s' => [
+                    'description' => 'Current status of the application.',
+                    'enum' => ['approved', 'declined', 'pending']
+                ],
+                'insertIPAddress:s' => 'IP address of the user who created the application.',
+                'dateInserted:dt' => 'When the application was created.'
+            ]);
+        }
+
+        return $fullSchema;
+    }
+
+    /**
+     * Get a list of current applicants.
+     *
+     * @return array
+     */
+    public function index(array $query) {
+        $this->permission('Garden.Users.Approve');
+
+        $in = $this->schema([
+            'page:i?' => [
+                'description' => 'Page number.',
+                'default' => 1,
+                'minimum' => 1,
+                'maximum' => 100
+            ],
+            'limit:i?' => [
+                'description' => 'The number of items per page.',
+                'default' => 30,
+                'minimum' => 1,
+                'maximum' => 100
+            ]
+        ], 'in')->setDescription('Get a list of current user applications.');
+        $out = $this->schema(
+            [':a' => $this->fullSchema()],
+            'out'
+        );
+
+        $query = $in->validate($query);
+
+        list($offset, $limit) = offsetLimit("p{$query['page']}", $query['limit']);
+        $rows = $this->userModel->getApplicants($limit, $offset)->resultArray();
+
+        foreach ($rows as &$row) {
+            $this->prepareRow($row);
+        }
+
+        $result = $out->validate($rows);
+        return $result;
+    }
+
+    /**
+     * Approve or decline an application.
+     *
+     * @param int $id
+     * @param array $body
+     * @throws ClientException if the user record is not an applicant.
+     * @return array
+     */
+    public function patch($id, array $body) {
+        $this->permission('Garden.Users.Approve');
+
+        $in = $this->schema([
+            'status:s' => [
```

Comment:
I'd like to avoid "user" references under the applications endpoint as much as we can. I know they are technically the same resource, but this endpoint aims to treat them as separate entities.

Reply:
Yeah that makes sense. I wrote this before realizing that the userID is actually the applicationID.

Still:
```
        $this->schema(['id' => 'Application ID'], 'in');
        $in = $this->schema([
            'status:s' => [
                'description' => 'Current status of the application.',
                'enum' => ['approved', 'declined', 'pending']
            ]
        ], 'in')->setDescription('Modify a user application.');
        $out = $this->schema($this->fullSchema(), 'out');
```